cli/cli-client/src/main/java/org/infinispan/cli/user/UserTool.java

```diff
@@ -0,0 +1,370 @@
+      }
+   }
+
+   private Encryption checkAlgorithm(Encryption encryption) {
+      if (encryption == Encryption.DEFAULT) {
+         // Not forcing anything, use what the current user.properties file specifies or the default
+         return this.encryption;
+      } else {
+         if (this.encryption == Encryption.DEFAULT) {
+            // We can override the default
+            return encryption;
+         } else if (this.encryption == encryption) {
+            // Compatible
+            return encryption;
+         } else {
+            throw MSG.userToolIncompatibleEncrypyion(encryption, this.encryption);
+         }
+      }
+   }
+
+   public String checkRealm(String realm) {
+      if (realm == null) {
+         return this.realm == null ? DEFAULT_REALM_NAME : this.realm;
+      } else {
+         if (this.realm == null || this.realm.equals(realm)) {
+            return realm;
+         } else {
+            throw MSG.userToolWrongRealm(realm, this.realm);
+         }
+      }
+   }
+
+   public void createUser(String username, String password, String realm, Encryption encryption, List<String> userGroups, List<String> algorithms) {
+      if (users.containsKey(username)) {
+         throw MSG.userToolUserExists(username);
+      }
+      realm = checkRealm(realm);
+      users.put(username, Encryption.CLEAR.equals(encryption) ? password : encryptPassword(username, realm, password, algorithms));
+      groups.put(username, userGroups != null ? String.join(",", userGroups) : "");
+      store(realm, encryption);
+   }
+
+   public String describeUser(String username) {
+      if (users.containsKey(username)) {
+         String[] userGroups = groups.containsKey(username) ? groups.getProperty(username).trim().split("\\s*,\\s*") : new String[]{};
+         return MSG.userDescribe(username, realm, userGroups);
+      } else {
+         throw MSG.userToolNoSuchUser(username);
+      }
+   }
+
+   public void removeUser(String username) {
+      users.remove(username);
+      groups.remove(username);
+      store();
+   }
+
+   public void modifyUser(String username, String password, String realm, Encryption encryption, List<String> userGroups, List<String> algorithms) {
+      if (!users.containsKey(username)) {
+         throw MSG.userToolNoSuchUser(username);
+      } else {
+         realm = checkRealm(realm);
+         if (password != null) { // change password
+            users.put(username, Encryption.CLEAR.equals(encryption) ? password : encryptPassword(username, realm, password, algorithms));
+         }
+         if (userGroups != null) { // change groups
+            groups.put(username, String.join(",", userGroups));
+         }
+         store(realm, encryption);
+      }
+   }
+
+   public void encryptAll(List<String> algorithms) {
+      if (this.encryption == Encryption.CLEAR) {
+         users.replaceAll((u, p) -> encryptPassword((String) u, realm, (String) p, algorithms));
+         store(realm, encryption);
```

Comment:
shouldn't be `Encryption.ENCRYPTED`?
```suggestion
         store(realm, Encryption.ENCRYPTED);
```